src/Fork.php

```diff
@@ -80,13 +80,13 @@ public function wait($pid = null)
             unset($this->threads[$pid]);
         }
 
+        $exceptions = $this->adapter->getExceptions();
```

Comment:
@eaglewu It's a little odd to call this here, what do you think about handling it in [SharedMemory](https://github.com/duncan3dc/fork-helper/blob/a253b551dfd98ef70307a2858c6b39cab4431235/src/SharedMemory.php#L8) using a [destructor](http://php.net/manual/en/language.oop5.decon.php#language.oop5.decon.destructor)?